code/harry_potter_store/test/test_store.py

```diff
@@ -0,0 +1,78 @@
+import unittest
+
+from harry_potter_store.src.store import Store
+
+
+class TestMain(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.store = Store()
+
+    def test_store_contain_five_book_types(self):
+        self.assertEqual(5, len(Store.book_types))
+
+    def test_get_price_1(self):
+        self.assertEqual(8, self.store.get_price({"1": 1}))
+
+    def test_get_price_2(self):
+        self.assertEqual(16, self.store.get_price({"1": 2}))
+
+    def test_get_price_with_discount_2_books(self):
+        self.assertEqual(16 * 0.95, self.store.get_price({"1": 1, "2": 1}))
+
+    def test_get_price_with_discount_3_books(self):
+        self.assertEqual(24 * 0.9, self.store.get_price({"1": 1, "2": 1, "3": 1}))
+
+    def test_get_price_with_discount_4_books(self):
+        self.assertEqual(32 * 0.8, self.store.get_price({"1": 1, "2": 1, "3": 1, "4": 1}))
+
+    def test_get_price_with_discount_5_books(self):
+        self.assertEqual(40 * 0.75, self.store.get_price({"1": 1, "2": 1, "3": 1, "4": 1, "5": 1}))
+
+    def test_get_price_complex_1(self):
+        self.assertEqual(16 * 0.95 + 8, self.store.get_price({"1": 1, "2": 2}))
+
+    def test_get_price_complex_2(self):
+        self.assertRaises(ValueError, lambda: self.store.get_price({"1": 0, "2": 2}))
+
+    def test_get_price_complex_3(self):
+        self.assertRaises(ValueError, lambda: self.store.get_price({"1": -132, "2": -3324}))
+
+    def test_get_price_complex_4(self):
+        self.assertRaises(ValueError, lambda: self.store.get_price({"1": 2, "2": 2, "111": 333}))
+
+    def test_get_price_complex_5(self):
+        self.assertEqual(24 * 0.90 + 16 * 0.95 + 8, self.store.get_price({"1": 3, "2": 1, "3": 2}))
+
+    def test_get_price_complex_6(self):
+        self.assertEqual((32 * 0.8) * 2, self.store.get_price({"1": 2, "2": 2, "3": 2, "4": 2}))
+
+    def test_get_price_complex_7(self):
+        self.assertEqual((40 * 0.75) + (24 * 0.9),
```

Comment:
So you went back to greedy algorithm. Then again, please add missing condition to the problem statement to make it clear which problem are you solving. For the current problem statement this is IMO wrong as it should be `book_price * 4 * 0.8 * 2` because 51.2 is less than 51.6. Same as case 10.

And if you went back to greedy algorithm, then what's the point of the previous patch where you rewrote straightforward greedy algorithm to the one with predicates and labmdas?

Reply:
yep, i have had an incorrect sorting
fixed it